tests/CMakeLists.txt

```diff
@@ -106,6 +106,8 @@ add_net_test_case(test_s3_upload_part_message_without_content_md5)
 add_net_test_case(test_s3_create_multipart_upload_message_with_content_md5)
 add_net_test_case(test_s3_complete_multipart_message_with_content_md5)
 add_net_test_case(test_s3_put_object_double_slashes)
+add_net_test_case(test_s3_put_object_no_content_length)
+add_net_test_case(test_s3_put_object_single_part_no_content_length)
```

Comment:
make sure we have tests that:
- create a 0-size Object
- create a noop part, after creating 1+ normal parts

Reply:
test for 0 size object added. it uncovered some issues that were fixed.
existing tests already hit path for noop part, due to tests having at most 3 parts and optimistic scheduling was set at min of 5